authors/apps/profiles/views.py

```diff
@@ -0,0 +1,74 @@
+from rest_framework.generics import ListAPIView, UpdateAPIView, RetrieveAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.exceptions import NotFound, PermissionDenied
+
+from .models import Profile
+from .serializers import UserProfileSerializer, UpdateUserProfileSerializer
+from .renderers import ProfileJSONRenderer
+
+class UserProfileView(RetrieveAPIView):
+    """ 
+    retrieves an authors profile information
+
+    """
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ProfileJSONRenderer,)
+    serializer_class = UserProfileSerializer
+
+    def get_queryset(self):
+        try:
+            profile = Profile.objects.select_related('user').get(
+                user__username=self.kwargs.get('username')
+            )
+            return profile
+        except Exception:
+
+            raise NotFound('User profile not found')
+
+    def retrieve(self, request, **kwargs):
+        data = self.get_queryset()
+        serializer = self.serializer_class(data, context={'request': request})
+
+        return Response(serializer.data)
+
+class AuthorsProfileListAPIView(ListAPIView):
```

Comment:
Good job but what is the relevance of this class? I think it would be better if users could also see their profile.
what was the thought process for this class?

Reply:
we saw it unnecessary for a user to see their profile when they  want to view a list of profiles.